### PR TITLE
added proposed default.gitignore

### PR DESCRIPTION
**Pull request by wz-gsa**
## Changes proposed in this pull request:

- added proposed default.gitignore for use across the organization. 

## Things to check
- Are we missing any potential useful defaults? 
- Will this cause any breakages?
- 
## Security considerations

This should improve security by ensuring developer configs and other files do not accidentally make their way into github.

### CODE REVIEW

File: .github/default.gitignore [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+.rspec
+.ruby-version
+.ruby-gemset
+# Ignore Byebug command history file.
+.byebug_history
+
+# Cloud Foundry specific
+manifest.yml
+manifest-*.yml
+*.env
```
Comment (bengerman13) [inline .github/default.gitignore:158]: I think these will cause problems, for example:
[we have a `manifest.yml` here])https://github.com/cloud-gov/cf-hello-worlds/blob/8a17d850b6cf13e679754da0adfc0b62c5245fcb/go-hello/manifest.yml#L4)

Including manifests is a somewhat-common pattern, and there are definitely ways to do it safely

File: .github/default.gitignore [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+*.vars.yml
+*.vars-*.yml
```
Comment (bengerman13) [inline .github/default.gitignore:160]: these _feel_ a little aggressive, especially with the globs. If my repo _needs_ a `${env}.vars.yml`, I'm likely to include an `example.vars.yml` or `dev.vars.yml.example`

File: .github/default.gitignore [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+/.dev_builds/
+/blobs/
+/config/
+/private.yml
+*.tgz
+
+# Concourse specific
+/fly-*
+/concourse-*
+/pipelines/
```
Comment (bengerman13) [inline .github/default.gitignore:176]: pretty sure we use this pattern deliberately somewhere?

File: .github/default.gitignore [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+*.sql
+*.sqlite
+*.sqlite3
+/log/
+*.env*
+**/log/*.log
+**/log/*.log*
+
+# OS Generated Files
+.DS_Store
```
Comment (bengerman13) [inline .github/default.gitignore:192]: 👨🏻‍🍳🤌

**Review by bengerman13 — COMMENTED** [review]
I have mixed feelings about this - having good defaults is awesome, but I don't know what overriding them looks like in this case. 

If we go the route of having this as a default repo-local gitignore, then we can of course edit as necessary (but we don't get protection on non-cg repos)
If we _suggest_ this as a global gitignore, then again, folks can edit as necessary, but it gets pretty weak as time and drift go on
If we _enforce_ this as a global gitignore, it seems likely that some of these will be surprising, so there will probably be some painful adjustment and tuning time

File: .github/default.gitignore (hunk 2 above)
Comment (wz-gsa) [reply .github/default.gitignore:160]: I think a `dev.vars.yml.example` would pass this, but you're right the other formats wouldn't. Having the examples format narrowly scoped/standardized is a good thing though imo. What do you think?

File: .github/default.gitignore (hunk 3 above)
Comment (cweibel) [reply .github/default.gitignore:176]: Everything I've seen has been /ci

Plus this only impacts new repos so 🤷

**Pull request closed without merging**